File: polyglot/runtime_template_resolver/mjs/src/interfaces.ts
import { LoggerInterface } from './logger.js';

export enum MissingStrategy {
    KEEP = 'KEEP',
    EMPTY = 'EMPTY',
    ERROR = 'ERROR',
    DEFAULT = 'DEFAULT',
}

export interface IResolverOptions {
    missingStrategy?: MissingStrategy;
    throwOnError?: boolean;
    logger?: LoggerInterface;
}

export interface IResolverResult {
    value: string;
    placeholders: string[];
    errors: Error[];
}

export interface ITemplateResolver {
    resolve(template: string, context: Record<string, unknown>, options?: IResolverOptions): string;
    resolveObject(obj: unknown, context: Record<string, unknown>, options?: IResolverOptions): unknown;
}

File: polyglot/runtime_template_resolver/py/src/runtime_template_resolver/interfaces.py
from dataclasses import dataclass, field
from typing import Any, Dict, List, Optional, Protocol, Union
from enum import Enum
from .logger import LoggerProtocol

class MissingStrategy(str, Enum):
    KEEP = "KEEP"
    EMPTY = "EMPTY"
    ERROR = "ERROR"
    DEFAULT = "DEFAULT"

@dataclass
class ResolverOptions:
    missing_strategy: Optional[MissingStrategy] = None
    throw_on_error: Optional[bool] = None
    logger: Optional[LoggerProtocol] = None

@dataclass
class ResolverResult:
    value: str
    placeholders: List[str] = field(default_factory=list)
    errors: List[Exception] = field(default_factory=list)

class TemplateResolverProtocol(Protocol):
    def resolve(self, template: str, context: Dict[str, Any], options: Optional[ResolverOptions] = None) -> str: ...
    def resolve_object(self, obj: Any, context: Dict[str, Any], options: Optional[ResolverOptions] = None) -> Any: ...
